src/components/Filter/Filter.js

```diff
@@ -1,36 +1,56 @@
 import React, { useState, useEffect } from "react";
-import { useParams, NavLink } from "react-router-dom";
+import { useParams, Link } from "react-router-dom";
 import axios from "axios";
 import Container from "@mui/material/Container";
 import Box from "@mui/material/Box";
 import Grid from "@mui/material/Grid";
-import ReusableCard from "../../components/ReusableCard/ReusableCard";
-import styled from "styled-components";
 import Typography from "@mui/material/Typography";
+import Pagination from "@mui/material/Pagination";
+import Stack from "@mui/material/Stack";
+import styled from "styled-components";
+import ReusableCard from "../../components/ReusableCard/ReusableCard";
+import ReusablePagination from "../Pagination/ReusablePagination";
+import All from "../../assets/Images/all.png";
+import African from "../../assets/Images/african.png";
+import American from "../../assets/Images/american.png";
+import Chinese from "../../assets/Images/chinese.png";
+import European from "../../assets/Images/european.png";
+import French from "../../assets/Images/french.png";
+import Indian from "../../assets/Images/indian.png";
+import Italian from "../../assets/Images/italian.png";
+import Korean from "../../assets/Images/korean.png";
+import Mexican from "../../assets/Images/mexican.png";
+import Thai from "../../assets/Images/thai.png";
 
 const Filter = () => {
   const [filterResult, setfilterResult] = useState([]);
+  const [page, setPage] = useState(1);
+
   const url = "/api/v1/recipes";
   let filter = useParams();
-  console.log(filter)
+  console.log(filter);
   const token = localStorage.getItem("myKitchenAppToken");
+  const perPage = 6; //number of recipes on each page
 
   const cuisineType = async () => {
     console.log(filter.intolerances, "we got here");
```

Comment:
Intolerance comes as undefined. I do not know why, but in this case, no recipes were returned. Screen shot is from dev-console after clicking one the filter options. For example; chinesee
<img width="607" alt="Screen Shot 2022-12-05 at 8 06 13 PM" src="https://user-images.githubusercontent.com/88150284/205811344-9dab90f7-aead-480a-83c2-bdfd9ab616bb.png">
<img width="564" alt="Screen Shot 2022-12-05 at 8 05 04 PM" src="https://user-images.githubusercontent.com/88150284/205811367-ed08e257-dfdb-4928-a913-21398f086ce7.png">

Reply:
yes, intolerances are returned as undefined if none of them is chosen. No recipes are returned because there are no recipes. That's because the search by ingredients is very strict. For example, when you are searching for "chicken" it will return you recipes with only "chicken" ingredients, not with "chicken breast" or "chicken breast halves"